Cargo cann't be built in msys2.

Cargo cann't download snapshot in msys2 and fails to build:

```
$ make
/mingw64/bin/python src/etc/dl-snapshot.py x86_64-pc-windows-gnu
Traceback (most recent call last):
  File "src/etc/dl-snapshot.py", line 80, in <module>
    with urlopen(url) as in_file:
  File "src/etc/dl-snapshot.py", line 14, in <lambda>
    urlopen = lambda url: closing(urlopen2(url))
  File "E:/msys64/mingw64/lib/python2.7/urllib2.py", line 154, in urlopen
    return opener.open(url, data, timeout)
  File "E:/msys64/mingw64/lib/python2.7/urllib2.py", line 431, in open
    response = self._open(req, data)
  File "E:/msys64/mingw64/lib/python2.7/urllib2.py", line 454, in _open
    'unknown_open', req)
  File "E:/msys64/mingw64/lib/python2.7/urllib2.py", line 409, in _call_chain
    result = func(*args)
  File "E:/msys64/mingw64/lib/python2.7/urllib2.py", line 1265, in unknown_open
    raise URLError('unknown url type: %s' % type)
urllib2.URLError: <urlopen error unknown url type: https>
Makefile:81: recipe for target 'target/snapshot/bin/cargo.exe' failed
make: *** [target/snapshot/bin/cargo.exe] Error 1
```

I'm stuck with similar situation and am a newbie when it comes to command line scripts. I am using a mac with OSX. Below is the exact error I get  when I run `make`. What should I do to get this up and running?

    raceback (most recent call last):ot.py 
    File "src/etc/dl-snapshot.py", line 34, in <module>
    vendor = ts[1]
    IndexError: list index out of range
    make: *** [target/snapshot/bin/cargo] Error 1

--- a/configure
+++ b/configure
@@ -264,6 +264,7 @@ need_cmd sed
 need_cmd file
 need_cmd cmake
 need_cmd pkg-config
+need_cmd curl
 
 CFG_SRC_DIR="$(cd $(dirname $0) && pwd)/"
 CFG_BUILD_DIR="$(pwd)/"

--- a/src/etc/dl-snapshot.py
+++ b/src/etc/dl-snapshot.py
@@ -1,20 +1,13 @@
 import distutils.spawn
 import hashlib
 import os
+import subprocess
 import sys
 import tarfile
 import shutil
 
-try:
-    from urllib.request import urlopen
-except ImportError:
-    # We are in python2
-    from urllib2 import urlopen as urlopen2
-    from contextlib import closing
-    urlopen = lambda url: closing(urlopen2(url))
-
-with open('src/snapshots.txt') as f:
-    lines = f.readlines()
+f = open('src/snapshots.txt')
+lines = f.readlines()
 
 date = lines[0]
 linux32 = lines[1]
@@ -64,7 +57,7 @@ if me is None:
 
 triple = new_triple
 
-platform, hash = me.strip().split()
+platform, hash = me.strip().split(' ')
 
 tarball = 'cargo-nightly-' + triple + '.tar.gz'
 url = 'https://static-rust-lang-org.s3.amazonaws.com/cargo-dist/' + date.strip() + '/' + tarball
@@ -77,22 +70,22 @@ if not os.path.isdir('target/dl'):
 if os.path.isdir(dst):
     shutil.rmtree(dst)
 
-with urlopen(url) as in_file:
-    data = in_file.read()
-    h = hashlib.sha1(data).hexdigest()
-    if h != hash:
-        raise Exception("failed to verify the checksum of the snapshot")
-    with open(dl_path, 'wb') as out_file:
-        out_file.write(data)
+ret = subprocess.call(["curl", "-o", dl_path, url])
+if ret != 0:
+    raise Exception("failed to fetch url")
+h = hashlib.sha1(open(dl_path, 'rb').read()).hexdigest()
+if h != hash:
+    raise Exception("failed to verify the checksum of the snapshot")
 
-with tarfile.open(dl_path) as tar:
-    for p in tar.getnames():
-        name = p.replace("cargo-nightly-" + triple + "/", "", 1)
-        fp = os.path.join(dst, name)
-        print("extracting " + p)
-        tar.extract(p, dst)
-        tp = os.path.join(dst, p)
-        if os.path.isdir(tp) and os.path.exists(fp):
-            continue
-        shutil.move(tp, fp)
+tar = tarfile.open(dl_path)
+for p in tar.getnames():
+    name = p.replace("cargo-nightly-" + triple + "/", "", 1)
+    fp = os.path.join(dst, name)
+    print("extracting " + p)
+    tar.extract(p, dst)
+    tp = os.path.join(dst, p)
+    if os.path.isdir(tp) and os.path.exists(fp):
+        continue
+    shutil.move(tp, fp)
+tar.close()
 shutil.rmtree(os.path.join(dst, 'cargo-nightly-' + triple))

--- a/src/etc/dl-snapshot.py
+++ b/src/etc/dl-snapshot.py
@@ -6,8 +6,8 @@ import sys
 import tarfile
 import shutil
 
-f = open('src/snapshots.txt')
-lines = f.readlines()
+with open('src/snapshots.txt') as f:
+    lines = f.readlines()
 
 date = lines[0]
 linux32 = lines[1]
@@ -57,7 +57,7 @@ if me is None:
 
 triple = new_triple
 
-platform, hash = me.strip().split(' ')
+platform, hash = me.strip().split()
 
 tarball = 'cargo-nightly-' + triple + '.tar.gz'
 url = 'https://static-rust-lang-org.s3.amazonaws.com/cargo-dist/' + date.strip() + '/' + tarball
@@ -77,15 +77,14 @@ h = hashlib.sha1(open(dl_path, 'rb').read()).hexdigest()
 if h != hash:
     raise Exception("failed to verify the checksum of the snapshot")
 
-tar = tarfile.open(dl_path)
-for p in tar.getnames():
-    name = p.replace("cargo-nightly-" + triple + "/", "", 1)
-    fp = os.path.join(dst, name)
-    print("extracting " + p)
-    tar.extract(p, dst)
-    tp = os.path.join(dst, p)
-    if os.path.isdir(tp) and os.path.exists(fp):
-        continue
-    shutil.move(tp, fp)
-tar.close()
+with tarfile.open(dl_path) as tar:
+    for p in tar.getnames():
+        name = p.replace("cargo-nightly-" + triple + "/", "", 1)
+        fp = os.path.join(dst, name)
+        print("extracting " + p)
+        tar.extract(p, dst)
+        tp = os.path.join(dst, p)
+        if os.path.isdir(tp) and os.path.exists(fp):
+            continue
+        shutil.move(tp, fp)
 shutil.rmtree(os.path.join(dst, 'cargo-nightly-' + triple))

